dcpy/models/product/dataset/metadata.py

```diff
@@ -245,3 +248,183 @@ def validate(self):
             set(col_names)
         ), "There should be no duplicate column names"
         # TODO: all the rest
+
+    def upgrade_to_v2(self):
```

Comment:
This is the fn that will convert v1 metadata to v2. We'll eventually just delete this entire file, so it's maybe less important to review every line of this, as opposed to just spot-checking the upgraded metadata

Reply:
another note for the possibility of `v` being 0 for the falsey function below